### PR TITLE
Update Mochi VAE parallelization scheme

### PR DESCRIPTION
**Pull request by sosborne-TT**
### Ticket
[40480](https://github.com/tenstorrent/tt-metal/issues/40480)

### Problem description
NeighborPad produces incorrect halo exchanges when both H-parallel and W-parallel dimensions are mapped to the same axis, since support was removed in PR [39739](https://github.com/tenstorrent/tt-metal/pull/39739). This results in PCC failures in Mochi VAE tests on Galaxy which uses an 8×4 mesh.

### What's changed
For mesh shapes 1xN, e.g. on T3K (1x8), N300 (1x2), N150 (1x1), use H/W parallel factor=1. For other mesh shapes, e.g. 8x4, 4x8, or 2x4 on Galaxy, parallelize H on axis 0 and W on axis 1.

### Checklist

  - [ ] [Mochi Galaxy integration test](https://github.com/tenstorrent/tt-metal/actions/runs/24912232240/job/72957518714) Hardware issue in CI -- tests were separately verified on a BH GLX machine.
  - [x] [Mochi Galaxy demo test](https://github.com/tenstorrent/tt-metal/actions/runs/25016985515/job/73268667671)
  - [x] [Mochi T3K integration test](https://github.com/tenstorrent/tt-metal/actions/runs/25086329740/job/73503326809)
  - [x] [Mochi T3K demo test](https://github.com/tenstorrent/tt-metal/actions/runs/25016953593/job/73268246827)

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:sosborne/mochi-test-fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:sosborne/mochi-test-fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:sosborne/mochi-test-fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:sosborne/mochi-test-fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:sosborne/mochi-test-fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:sosborne/mochi-test-fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:sosborne/mochi-test-fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:sosborne/mochi-test-fix)

### CODE REVIEW

File: models/tt_dit/tests/models/mochi/test_vae_mochi.py [hunk 1]
```diff
@@ -75,87 +91,71 @@ def create_random_conv3d_models(mesh_device, in_channels, out_channels, bias=Tru
...
+        "2x4",
+        "wh_8x4",
+        "bh_8x4",
+    ],
+    indirect=["mesh_device"],
+)
+@pytest.mark.parametrize(
+    "device_params", [{"fabric_config": ttnn.FabricConfig.FABRIC_1D, "trace_region_size": 20000000}], indirect=True
+)
+def test_tt_conv3d_1x1x1(mesh_device, N, C_in, C_out, T, H, W, reset_seeds, num_links):
```
[the remark this answers is not in the document]
Comment (sosborne-TT) [reply models/tt_dit/tests/models/mochi/test_vae_mochi.py]: Addressed

File: tests/scripts/t3000/run_t3000_integration_tests.sh [hunk 2]
```diff
@@ -408,7 +408,7 @@ run_t3000_mochi_tests() {
   echo "LOG_METAL: Running run_t3000_mochi_tests"
 
   export TT_DIT_CACHE_DIR="/tmp/TT_DIT_CACHE"
-  FAKE_DEVICE=T3K pytest models/tt_dit/tests/models/mochi/test_vae_mochi.py -k "decoder and 1link-load_dit-small_latent or conv3d_1x1x1 or -1link-l768" --timeout=1500; fail+=$?
+  FAKE_DEVICE=T3K pytest models/tt_dit/tests/models/mochi/test_vae_mochi.py -k "decoder and 2x4-load_dit-small_latent or conv3d_1x1x1 or -2x4-l768" --timeout=1500; fail+=$?
```
[the remark this answers is not in the document]
Comment (sosborne-TT) [reply tests/scripts/t3000/run_t3000_integration_tests.sh]: Addressed

File: tests/scripts/t3000/run_t3000_unit_tests.sh [hunk 3]
```diff
@@ -323,7 +323,7 @@ run_t3000_tt_dit_tests() {
   DIT_UNIT_TEST=1 pytest models/tt_dit/tests/models/mochi/test_transformer_mochi.py::test_mochi_transformer_model[wormhole_b0-device_params0-no_load_cache-no_test_attention_mask-short_seq-2x4sp0tp1-True] ; fail+=$?
 
   #Mochi VAE main component
-  FAKE_DEVICE=T3K pytest models/tt_dit/tests/models/mochi/test_vae_mochi.py::test_tt_resblock_forward[wormhole_b0-mesh_device0-device_params0-1link-l768] ; fail+=$?
+  FAKE_DEVICE=T3K pytest models/tt_dit/tests/models/mochi/test_vae_mochi.py -k "resblock_forward and 2x4 and l768 and bf16" ; fail+=$?
```
[the remark this answers is not in the document]
Comment (sosborne-TT) [reply tests/scripts/t3000/run_t3000_unit_tests.sh]: Addressed

**Review by jonathansuTT — APPROVED** [review]

**Review by cglagovichTT — APPROVED** [review]

**Review by jyuanTT — APPROVED** [review]
